sharedcache/OSCacheConfigOptions.hpp

```diff
@@ -0,0 +1,286 @@
+
+    /** 
+     * @brief Associated with the StatsReason enum. Returns false for StatReason_List.
+     * @return bool
+     */ 
+    virtual bool omr_statList() = 0;
+
+    /** 
+     * @brief Sets the base directory that needs to be appended to the filename.
+     * @param[in] appendBaseDir Append Base Directory
+     * @return OSCacheConfigOptions&
+     */ 
+    virtual OSCacheConfigOptions& omr_setAppendBaseDir(bool appendBaseDir) = 0;
+
+    /** 
+     * @brief Checks if semaphore check is enabled
+     * @return bool
+     */ 
+    virtual bool omr_semaphoreCheckEnabled() = 0;
+
+    /** 
+     * @brief Sets the StatReason members
+     * @param[in] reason Reason value from the StatReason enum 
+     * @return OSCacheConfigOptions&
+     */ 
+    virtual OSCacheConfigOptions& omr_setStatReason(StatReason reason) = 0;
+
+    /** 
+     * @brief Sets to StartupReason_Open of StartupReason enum 
+     * @param[in] reason Reason value from the StartupReason enum 
+     * @return OSCacheConfigOptions&
+     */ 
+    virtual OSCacheConfigOptions& omr_setOpenReason(StartupReason reason) = 0;
+
+    /** 
+     * @brief Sets to the cache to Read-Only open mode
+     * @return OSCacheConfigOptions&
+     */ 
+    virtual OSCacheConfigOptions& omr_setReadOnlyOpenMode() = 0;
+
+    /** 
+     * @brief Sets the OS cache to open mode 
+     * @param[in] openMode Open Mode
+     * @return OSCacheConfigOptions&
+     */ 
+    virtual OSCacheConfigOptions& omr_setOpenMode(I_32 openMode) = 0;
+
+     /** 
+     * @brief Returns the block size of the cache
+     * @return OSCacheConfigOptions&
+     */ 
+    virtual U_32 omr_cacheSize() = 0;
+
+     /** 
+     * @brief sets the Cache Size
+     * @param[in] size New Size
+     * @return OSCacheConfigOptions&
+     */ 
+    virtual OSCacheConfigOptions& omr_setCacheSize(U_32 size) = 0;
+
+     /** 
+     * @brief Gets the maxCRCSamples value from the CacheCRCChecker
+     * @return U_32 _maxCRCSamples value
+     */ 
+    virtual U_32 omr_maxCRCSamples() = 0;
+
+     /** 
+     * @brief To check if the cache creates a file
+     * @return bool
+     */ 
+    virtual bool omr_createFile() = 0;
+
+     /** 
+     * @brief To check if we try to open the cache read-only when or if we failed to open the cache with write permissions
+     * @return bool
+     */ 
+    virtual bool omr_tryReadOnlyOnOpenFailure() = 0;
+
+     /** 
+     * @brief To check if we dumo the contents when we cache is corrupt
+     * @return bool
+     */ 
+    virtual bool omr_disableCorruptCacheDumps() = 0;
+
+     /** 
+     * @brief To check if the VERBOSE flag is enabled
+     * @return bool
+     */ 
+    virtual bool omr_verboseEnabled() = 0;
+
+     /** 
+     * @brief To check if the group access is enabled
+     * @return bool
+     */ 
+    virtual bool omr_groupAccessEnabled() = 0;
+
+     /** 
+     * @brief Renders the options to a bit vector understood by the functions of the OMR port library.
+     * @return bool
```

Comment:
Return type is `U_32`, not `bool`. 

The comments for return type are all wrong for the following functions in this file.